crates/noirc_evaluator/src/ssa_refactor/opt/inlining.rs

```diff
@@ -73,8 +73,13 @@ struct PerFunctionContext<'function> {
     /// argument values.
     values: HashMap<ValueId, ValueId>,
 
-    /// Maps BasicBlockIds in the function being inlined to the new BasicBlockIds to use in the
-    /// function being inlined into.
+    /// Maps BasicBlockIds in the function being inlined from to the new BasicBlockIds to use in
+    /// the function being inlined into.
+    ///
+    /// Note that this 1-to-1 mapping only holds true in respect to the start of a source block.
+    /// This is because the inlining process potentially splits a single block into many. It is
+    /// not safe to query this mapping in respect to a program point midway or at the end of a
+    /// source block because the mapping may be stale.
```

Comment:
Wording: I would say it is still safe, but we should change expectations instead. Since it is not actually a 1-to-1 mapping we should instead specify it is a 1 to many mapping, where the block returned for a given key is always the start of a set of possibly multiple blocks.